notebooks/DATB/psf_grid_simulation_photometry/GriddedPSFModel_simulation_and_photometry.ipynb

```diff
@@ -0,0 +1,471 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's display the simulated JWST NIRCam det A1 F090W image."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "norm = simple_norm(data, 'sqrt', percent=99.)\n",
+    "plt.figure(figsize=(15, 15))\n",
+    "plt.imshow(data, norm=norm, origin='lower')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### PSF-fitting photometry with `GriddPSFModel`"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can also use `GriddedPSFModel` to perform PSF-fitting photometry.\n",
+    "Let's start by creating another simulated image with only a few sources, which \n",
+    "will enable the fitting to run much faster."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "shape = (2047, 2047)\n",
+    "data = np.zeros(shape)\n",
+    "nstars = 25\n",
+    "from astropy.utils.misc import NumpyRNGContext\n",
+    "with NumpyRNGContext(12345):    # seed for repeatability\n",
+    "    xx = np.random.uniform(low=0, high=shape[1], size=nstars)\n",
+    "    yy = np.random.uniform(low=0, high=shape[0], size=nstars)\n",
+    "    zz = np.random.uniform(low=0, high=200000., size=nstars)\n",
+    "    \n",
+    "eval_xshape = np.int(np.ceil(psfmodel.data.shape[2] / psfmodel.oversampling))\n",
+    "eval_yshape = np.int(np.ceil(psfmodel.data.shape[1] / psfmodel.oversampling))\n",
+    "\n",
+    "for xxi, yyi, zzi in zip(xx, yy, zz):\n",
+    "    x0 = np.int(np.floor(xxi - (eval_xshape - 1) / 2.))\n",
+    "    y0 = np.int(np.floor(yyi - (eval_yshape - 1) / 2.))\n",
+    "    x1 = x0 + eval_xshape\n",
+    "    y1 = y0 + eval_yshape\n",
+    "    \n",
+    "    if x0 < 0:\n",
+    "        x0 = 0\n",
+    "    if y0 < 0:\n",
+    "        y0 = 0\n",
+    "    if x1 > shape[1]:\n",
+    "        x1 = shape[1]\n",
+    "    if y1 > shape[0]:\n",
+    "        y1 = shape[0]\n",
+    "        \n",
+    "    y, x = np.mgrid[y0:y1, x0:x1]\n",
+    "    data[y, x] += psfmodel.evaluate(x=x, y=y, flux=zzi, x_0=xxi, y_0=yyi)\n",
+    "    \n",
+    "noise = make_noise_image(data.shape, 'gaussian', mean=0, stddev=2, random_state=123)\n",
+    "data += noise"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's view this new simulated image."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "norm = simple_norm(data, 'sqrt', percent=99.)\n",
+    "plt.figure(figsize=(15, 15))\n",
+    "plt.imshow(data, norm=norm, origin='lower')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For this simple example, we'll use `photutils.BasicPSFPhotometry` to perform the photometry."
```

Comment:
Maybe point out that "basic" is relevant here mainly because we're not using the finder?

Reply:
That and not using the iterative photometry (both for speed).

Reply:
Done.